propagation/b3/map_test.go

```diff
@@ -0,0 +1,408 @@
+// Copyright 2019 The OpenZipkin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package b3_test
+
+import (
+	"testing"
+
+	zipkin "github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/propagation/b3"
+	"github.com/openzipkin/zipkin-go/reporter/recorder"
+)
+
+func TestMapExtractFlagsOnly(t *testing.T) {
+	m := make(b3.Map)
+
+	m[b3.Flags] = "1"
+
+	sc, err := m.Extract()
+	if err != nil {
+		t.Fatalf("Extract failed: %+v", err)
+	}
+
+	if want, have := true, sc.Debug; want != have {
+		t.Errorf("sc.Debug want %+v, have %+v", want, have)
+	}
+}
+
+func TestMapExtractSampledOnly(t *testing.T) {
+	m := make(b3.Map)
+
+	m[b3.Sampled] = "0"
+
+	sc, err := m.Extract()
+	if err != nil {
+		t.Fatalf("Extract failed: %+v", err)
+	}
+
+	if sc.Sampled == nil {
+		t.Fatalf("Sampled want %t, have nil", false)
+	}
+
+	if want, have := false, *sc.Sampled; want != have {
+		t.Errorf("Sampled want %t, have %t", want, have)
+	}
```

Comment:
Just a comment, I'm surprised that go doesn't include matchers in the standard library test utilities